panther_battery/src/battery_node.cpp

```diff
@@ -58,6 +60,10 @@ void BatteryNode::Initialize()
     } catch (std::runtime_error & err) {
       RCLCPP_WARN(this->get_logger(), "Failed to initialize ADC Battery: %s", err.what());
       RCLCPP_INFO(this->get_logger(), "Using Roboteq drivers to publish battery data.");
+
+      diagnostic_updater_->broadcast(
+        diagnostic_msgs::msg::DiagnosticStatus::WARN,
+        "Failed to initialize ADC Battery: " + std::string(err.what()));
```

Comment:
I dig into this problem to find a solution. In general, we can't see the broadcasted message because the logic underneath sends the provided message on **all defined tasks**. And since at this point, we don't have any task defined (the `battery_publisher` adds one) we see no message.

The workaround would be to define a new additional task at this point to enable wanted broadcasting, but once I started thinking about it, it finally seemed pointless. Diagnostic tasks after all should be used for periodic diagnostics. In case of an initialization issue relevant to the node and not to other diagnostic tasks, it seems more natural to have an `RCLCPP_INFO/`WARN` for this message. 

My proposal here is to get rid of this broadcasted message in favor of the standard ROS logging. 

Let me think if u're ok with it.

Reply:
As we discussed, we stay with current implementation